### ISSUE TITLE
Scene Mount entity does not remember last location after game exit

### ISSUE THREAD

**Issue by ApolloGM** [suriyun-production/mmorpg-kit-docs#1340]
Mount resets to its original location once a player logs off.
To replicate in scene Map001 , you can test this by mounting the Alpaca 
ride the Alpaca to a different location
log off
log back in and the Mount Alpaca is back to its original location

**Comment by ApolloGM** [comment]
if there is a workaround or an easy fix I can implement, please advise. I need the mount to remember its last location. 
The behavior I am trying to accomplish is that a multi passenger train follows a route and makes stops along the way to pick up passengers (players). But when the players log off the train resets to its first stop. Is there any way to get the mount to remember its last location on the server?

**Comment by insthync** [comment]
No any easy fix :D

**Comment by ApolloGM** [comment]
> 

Is it possible at all? I am thinking of other workarounds. Maybe save the asset location on the server somehow? It would be a great help.

**Comment by ApolloGM** [comment]
tried a few things but didn't work. Please look into it whenever possible. It is really important for me to get this functionality working. Thanks again! I am willing to offer testing assistance and if you have a patreon I will donate. Thanks and keep up the great work!

**Comment by insthync** [comment]
If you are going to implement it for offline/LAN games, you have to add entity saving functionality to game save classes ( `BaseGameSaveSystem` / `DefaultGameSaveSystem`). And call load/save function in `LanRpgNetworkManager`

If you are going to do it for MMO games, you have to add the functionality to database classes  (`BaseDatabase` / `MySQLDatabase`) to handle database operations, also add the functionality to `DatabaseNetworkManager` which made for database server connection to make it able to connect to the database server and do stuff. And call load/save function in `MapNetworkManager`.

You can look at build save data as an example

If you want to donate, you can do it via https://ko-fi.com/insthync

**Comment by ApolloGM** [comment]
ok thanks, I will try and let you know. fyi I downloaded your new version with the in scene mount fix, but I get compile errors in 2021.2.5

**Comment by ApolloGM** [comment]
![mmorpgerror2021 2](https://user-images.githubusercontent.com/95550185/144908617-82db2e05-994c-49fe-bf50-7725d7c3da7e.jpg)

**Comment by insthync** [comment]
You do what is written in the last line?

**Comment by ApolloGM** [comment]
yes I imported and no more errors. Thank you!
Though the mount still gets destroyed on disconnect and resets location on game exit. 
I will keep trying.

**Comment by ApolloGM** [comment]
so I wrote my own entity location saving rigid body movement script. It now remembers the last location. I do have another related issue however. 

I removed the rigidbody movement script on the mount and added my own, but now the movement is not seen across the network by other players. I assumed that the LinenetLib component and player controller is responsible for the network sync.

 Is your rigidbody entity movement script on the mount also needed for network syncing?

 I need my own rigidbody movement so I can slerp the movement since it is a train (and turns like a bus). Your mount movement does not slerp, so it instantly turns with WASD. Any way to configure the slerp parameters in your rigidbody movement script?
